### PR TITLE
remove webSecurity:false in webPreferences

### PR DESCRIPTION
**Pull request by diracdeltas**
and add some comments for security-sensitive code

fix https://github.com/brave/browser-laptop/issues/10240

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: js/stores/appStore.js [hunk 1]
```diff
@@ -313,10 +313,9 @@ function windowDefaults () {
     minModalWidth: 100,
     windowOffset: 20,
     webPreferences: {
+      // XXX: Do not edit without security review
       sharedWorker: true,
-      nodeIntegration: false,
```
Comment (diracdeltas) [inline js/stores/appStore.js]: this is not needed anymore, right?

Comment (bridiver) [inline js/stores/appStore.js]: yea, does nothing

File: app/browser/tabs.js [hunk 2]
```diff
@@ -706,6 +706,7 @@ const api = {
   },
 
   executeScriptInBackground: (script, cb) => {
+    // Do not edit without security review
```
Comment (bridiver) [inline app/browser/tabs.js]: this is not a web-accessible api and is only used in the browser process. Actually now that we have worker threads it isn't used at all

File: js/stores/appStore.js [hunk 3]
```diff
@@ -313,10 +313,9 @@ function windowDefaults () {
...
       partition: 'default',
-      webSecurity: false,
       allowFileAccessFromFileUrls: true,
```
Comment (bridiver) [inline js/stores/appStore.js]: I don't know if these are necessary anymore either since we're using `chrome://brave` urls now

**Comment by diracdeltas** [comment]
@bridiver i reverted https://github.com/brave/browser-laptop/commit/42aa0cc6515725151ecfb97c40fb6f4ffb0892cf. can't tell if anything is broken. please let me know if you remember a test case.

**Comment by diracdeltas** [comment]
@bridiver ready for review

File: app/browser/tabs.js [hunk 4]
```diff
@@ -699,22 +698,6 @@ const api = {
     })
   },
 
-  executeScriptInBackground: (script, cb) => {
```
Comment (bridiver) [inline app/browser/tabs.js]: I think this is being used in a PR in progress

Comment (NejcZdovc) [reply app/browser/tabs.js]: Yes this is used in this PR #10325

**Comment by diracdeltas** [comment]
fixed the merge conflict. @NejcZdovc @bridiver could one of you review plz

**Comment by luixxiul** [comment]
no-qa-needed?

**Pull request merged**